Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -78,8 +79,9 @@ enum class SendOption : uint8_t {
     DispatchMessageEvenWhenWaitingForSyncReply = 1 << 0,
     DispatchMessageEvenWhenWaitingForUnboundedSyncReply = 1 << 1,
     IgnoreFullySynchronousMode = 1 << 2,
+    NoDispatchToMainOnError = 1 << 3,
```

Comment:
I'm not fan of introducing yet-another if arm. It'd be great if this discrepancy would be removed altogether and the semantics were well-defined from the start.
However, it's not very productive to require this for this patch..

Reply:
As discussed offline, I've removed this option and let the callback be dispatched through the main runloop on error
I also added two tests that checks that the CB is still called asynchronously as it's the intended behaviour.